src/lib_io.c

```diff
@@ -166,6 +166,15 @@ static void io_file_readall(lua_State *L, FILE *fp)
     char *buf = lj_buf_tmp(L, m);
     n += (MSize)fread(buf+n, 1, m-n, fp);
     if (n != m) {
+      if (ferror(fp) != 0 && errno == EINTR) {
+        do {
+          n += (MSize)fread(buf+n, 1, m-n, fp);
```

Comment:
We should call `clearerr()` to clear the file error indicator before this line because "The error indicator can be reset only by the clearerr() function." according to the manpage of `ferror()`. Otherwise we may have an infinite loop here.

Reply:
@agentzh 
My stupid mistake. I will fix it later.